relayer/src/rpc_conf.rs

```diff
@@ -0,0 +1,197 @@
+/*
+ For ApiKey data structure
+ */
+
+use std::fmt::Debug;
+use color_eyre::Report;
+
+#[derive(Eq, Hash, Clone, Debug, PartialEq)]
+pub struct ApiKey(pub near_jsonrpc_client::auth::ApiKey);
+
+impl From<ApiKey> for near_jsonrpc_client::auth::ApiKey {
+    fn from(api_key: ApiKey) -> Self {
+        api_key.0
+    }
+}
+
+impl std::fmt::Display for ApiKey {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+impl std::str::FromStr for ApiKey {
+    type Err = color_eyre::eyre::Report;
+
+    fn from_str(api_key: &str) -> Result<Self, Self::Err> {
+        Ok(Self(near_jsonrpc_client::auth::ApiKey::new(api_key)?))
+    }
+}
+
+impl serde::ser::Serialize for ApiKey {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: serde::ser::Serializer
+    {
+        serializer.serialize_str(&self.0.to_str().unwrap())
+    }
+}
+
+impl<'de> serde::de::Deserialize<'de> for ApiKey {
+    fn deserialize<D>(deserializer: D) -> Result<ApiKey, D::Error>
+        where
+            D: serde::de::Deserializer<'de>,
+    {
+        String::deserialize(deserializer)?
+            .parse()
+            .map_err(|err: color_eyre::eyre::Report| serde::de::Error::custom(err.to_string()))
+    }
+}
+
+/*
+ For RPCConfig data structure for creating the RPC connection
+ */
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct RPCConfig {
+    pub credentials_home_dir: std::path::PathBuf,
+    pub networks: linked_hash_map::LinkedHashMap<String, NetworkConfig>,
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct NetworkConfig {
+    pub network_name: String,
+    pub rpc_url: url::Url,
+    pub rpc_api_key: Option<ApiKey>,
+    pub wallet_url: url::Url,
+    pub explorer_transaction_url: url::Url,
+}
+
+impl Default for RPCConfig {
+    fn default() -> Self {
```

Comment:
This is more of a constructor than a default, especially since it is non-deterministic and can panic.

Reply:
I think it's OK to keep it as a default since the RPC can be contructed by values in the config like what can happen [in main.rs](https://github.com/near/pagoda-relayer-rs/pull/1/files#diff-42cb6807ad74b3e201c5a7ca98b911c5fa08380e942be6e4ac5807f8377f87fcR50) when `override_rpc_conf = true` in `config.toml`